cmd/cluster/ssh/key.go

```diff
@@ -0,0 +1,157 @@
+-----END RSA PRIVATE KEY-----
+
+Omitting the $CLUSTER_ID will print the ssh key for the cluster you're currently logged into.
+
+
+$ osdctl cluster ssh key > /tmp/ssh.key
+INFO[0005] Backplane URL retrieved via OCM environment: https://api.backplane.openshift.com
+$ cat /tmp/ssh.key
+-----BEGIN RSA PRIVATE KEY-----
+...
+-----END RSA PRIVATE KEY-----
+
+Despite the logs from backplane, the ssh key is the only output channelled through stdout. This means you can safely redirect the output to a file for greater convienence.`,
+		Args: cobra.MaximumNArgs(1),
+		RunE: func(_ *cobra.Command, args []string) error {
+
+			// If user provides an argument: use it to identify the cluster's hive shard,
+			// otherwise use the current cluster's ID
+			clusterID := ""
+			var err error
+			if len(args) == 0 {
+				clusterID, err = k8s.GetCurrentCluster()
+				if err != nil {
+					return fmt.Errorf("failed to retrieve ID for current cluster")
+				}
+			} else {
+				clusterID = args[0]
+			}
+
+			err = PrintKey(clusterID)
+			if err != nil {
+				return fmt.Errorf("failed to retrieve ssh key for cluster %s: %w", clusterID, err)
+			}
+			return nil
+		},
+	}
+	return cmd
+}
+
+// PrintKey retrieves the cluster's private ssh key from hive and prints it to stdout.
+func PrintKey(identifier string) error {
+	// Login to the provided cluster's hive shard
+	ocmClient, err := utils.CreateConnection()
+	if err != nil {
+		return fmt.Errorf("failed to establish connection to OCM: %w", err)
+	}
+
+	cluster, err := utils.GetCluster(ocmClient, identifier)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve cluster from OCM: %w", err)
+	}
+
+	clusterID := cluster.ID()
+	hive, err := utils.GetHiveCluster(clusterID)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve hive shard for cluster: %w", err)
+	}
+
+	scheme := runtime.NewScheme()
+	corev1.AddToScheme(scheme)
+	hiveClient, err := k8s.NewAsBackplaneClusterAdmin(hive.ID(), client.Options{Scheme: scheme})
+	if err != nil {
+		return fmt.Errorf("failed to create privileged client: %w", err)
+	}
+
+	// Determine the cluster's hive namespace via cluster ID
+	namespaces := corev1.NamespaceList{}
+	err = hiveClient.List(context.TODO(), &namespaces)
+	if err != nil {
+		return fmt.Errorf("failed to list hive namespaces: %w", err)
+	}
+
+	namespace, err := findClusterNamespace(namespaces, clusterID)
+	if err != nil {
+		return fmt.Errorf("failed to locate cluster namespace in hive: %w", err)
+	}
+
+	// Grab secret from the cluster's hive NS
+	secrets := corev1.SecretList{}
+	err = hiveClient.List(context.TODO(), &secrets, &client.ListOptions{Namespace: namespace.Name})
+	if err != nil {
+		return fmt.Errorf("failed to retrieve secrets from hive: %w", err)
+	}
+
+	secret, err := findSSHSecret(secrets)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve ssh key from hive: %w", err)
+	}
+
+	// Grab the correct file out of the secret & decode
+	encodedPrivateKey, found := secret.Data[privateKeyFilename]
```

Comment:
Isn't it decoded once you call `secret.Data[privateKeyFilename]`? Suggest renaming `encodedPrivateKey` to `decodedPrivateKey`

Reply:
No, [secret data](https://pkg.go.dev/k8s.io/api/core/v1#Secret.Data) is of type `map[string]byte`. The `encodedPrivateKey` is a byte array until it's cast as a string below